src/service/helper.ts

```diff
@@ -143,17 +143,40 @@ export const AdoApiUtil = {
   },
 
   async collectTeamDictionary(teams: WebApiTeam[], filter?: FilterInterface, clientOptions?: IVssRestClientOptions): Promise<Map<string, TeamDictionaryValue>> {
-    const workItemsClient = getClient(WorkItemTrackingRestClient, clientOptions);
+
     const teamWorkItems = await Promise.all(teams.map(it => fetchTeamWorkItems(it, filter, clientOptions)));
-    const projectItems: Map<string, TeamDictionaryValue>[] = await Promise.all(teamWorkItems.map(({ id, ids, connections }) => ids.length > 0 ? workItemsClient.getWorkItemsBatch({
-      $expand: WorkItemExpand.All,
-      asOf: new Date(),
-      errorPolicy: WorkItemErrorPolicy.Omit,
-      fields: [],
-      ids
-    }).then((items = []) => new Map<string, TeamDictionaryValue>([[id, { connections, map: new Map(items.map(item => [`${item.id}`, item])) }]]))
+    const projectItems: Map<string, TeamDictionaryValue>[] = await Promise.all(teamWorkItems.map(({ id, ids, connections }) => ids.length > 0 ? AdoApiUtil.getWorkItemsBatch(id, ids, connections, clientOptions)
       : Promise.resolve(new Map<string, TeamDictionaryValue>([[id, { connections: {}, map: new Map() }]]))));
 
     return projectItems.reduce((acc, next) => new Map([...acc, ...next]), new Map());
-  }
+  },
+
+  async getWorkItemsBatch(id: string, ids: number[], connections: { [key: string]: WorkItemLink[] }, clientOptions?: IVssRestClientOptions): Promise<Map<string, TeamDictionaryValue>> {
+    let batch = [];
+    if (ids.length > 200) {
+      let index = 0, chunk = 0, length = ids.length;
+      while (index < length) {
+        batch[chunk++] = ids.slice(index, (index += 200));
+      }
+    } else {
+      batch = [ids]
+    }
+
+    let items = [];
+    const workItemsClient = getClient(WorkItemTrackingRestClient, clientOptions);
```

Comment:
Hey!) Looks good to me - at least provided logic is working fine. But maybe it worth to replace loop of single async/await call to promise.all - hopefully we can parallelize request to some possible extent imposed by browser.

`await Promise.all(promises.map(it => it.then(data => { console.log(data); return data; }).catch(e => { console.error(e);  return []}))).reduce((acc, next) => [...acc, ...next], [])`

Reply:
done!